core/js/src/main/scala/cats/effect/IOApp.scala

```diff
@@ -178,6 +178,13 @@ trait IOApp {
   protected def reportFailure(err: Throwable): IO[Unit] =
     Console[IO].printStackTrace(err)
 
+  /**
+   * Defines what to do when CpuStarvationCheck is triggered. Defaults to log a warning to
+   * System.err.
+   */
+  protected def onCpuStarvationWarn: CpuStarvationWarningMetrics => IO[Unit] =
```

Comment:
Actually one quick nit: shouldn't this be a method, rather than a function?

```suggestion
  protected def onCpuStarvationWarn(metrics: CpuStarvationWarningMetrics): IO[Unit] =
```

It means we can't define it in a point-free fashion, but the users are much more likely to define this in a non-point-free fashion, similar to `reportFailure`.

Reply:
Will look into this tomorrow morning.  👍  Thanks for the review